clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -334,14 +334,15 @@ public static void sleep(long ms) {
      * Instantiate the class
      */
     public static <T> T newInstance(Class<T> c) {
+        if (c == null)
+            throw new NullPointerException();
```

Comment:
I thought it's better to validate the argument before trying to create an instance. And, the code seems unfit in the `try..catch` block.